src/rewrite.jl

```diff
@@ -291,114 +291,285 @@ function _write_add_mul(vectorized, minus, current_sum, left_factors, inner_fact
+                    new_var,
                 )
-                return new_var, Expr(:block, parsed, power_expr)
+                return new_var, Expr(:block, base_rewrite, new_expr)
             end
         elseif inner_factor.args[1] == :/ && !vectorized
+            # Rewrite
+            #   new_var = numerator / denominator
+            # as
+            #   new_var = numerator * (1 / denominator)
             @assert length(inner_factor.args) == 3
-            numerator = inner_factor.args[2]
-            denom = inner_factor.args[3]
-            return _rewrite(vectorized, minus, numerator, current_sum, left_factors, vcat(esc(:(1 / $denom)), right_factors), new_var)
-        elseif length(inner_factor.args) >= 2 && (isexpr(inner_factor.args[2], :generator) || isexpr(inner_factor.args[2], :flatten))
-            return new_var, _parse_generator(vectorized, minus, inner_factor, current_sum, left_factors, right_factors, new_var)
+            return _rewrite(
+                vectorized,
+                minus,
+                inner_factor.args[2],
+                current_sum,
+                left_factors,
+                vcat(esc(:(1 / $(inner_factor.args[3]))), right_factors),
+                new_var,
+            )
+        elseif (
+            length(inner_factor.args) >= 2 &&
+            (
+                isexpr(inner_factor.args[2], :generator) ||
+                isexpr(inner_factor.args[2], :flatten)
+            )
+        )
+            # A generator statement.
+            code = _parse_generator(
+                vectorized,
+                minus,
+                inner_factor,
+                current_sum,
+                left_factors,
+                right_factors,
+                new_var,
+            )
+            return new_var, code
         end
-    elseif isexpr(inner_factor, :curly)
-        Base.error("The curly syntax (sum{},prod{},norm2{}) is no longer supported. Expression: `$inner_factor`.")
     end
-    if isa(inner_factor, Expr) && _is_comparison(inner_factor)
+    if isexpr(inner_factor, :curly)
+        error(
+            "The curly syntax (sum{},prod{},norm2{}) is no longer supported. " *
+            "Expression: `$inner_factor`."
+        )
+    elseif isa(inner_factor, Expr) && _is_comparison(inner_factor)
         error("Unexpected comparison in expression `$inner_factor`.")
-    end
-    if isa(inner_factor, Expr) && _has_assignment_in_ref(inner_factor)
+    elseif isa(inner_factor, Expr) && _has_assignment_in_ref(inner_factor)
         error("Unexpected assignment in expression `$inner_factor`.")
     end
-    # at the lowest level
-    return new_var, _write_add_mul(vectorized, minus, current_sum, left_factors, (esc(inner_factor),), right_factors, new_var)
+    # None of the special cases were hit! This probably means we are vectorized.
+    code = _write_add_mul(
+        vectorized,
+        minus,
+        current_sum,
+        left_factors,
+        (esc(inner_factor),),
```

Comment:
The bug for 1.6 is here. If `inner_factor` contains a `sum(1 for _ = 1:0)`, then this evaluates it and throws an error. I'll think about what extra logic needs to be added to fix it in #60